splunk-otel-android/src/main/java/com/splunk/rum/reactnative/RNSpanData.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum.reactnative;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.internal.ImmutableSpanContext;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.trace.data.DelegatingSpanData;
+import io.opentelemetry.sdk.trace.data.SpanData;
+
+public final class RNSpanData extends DelegatingSpanData {
+
+    private final SpanContext modifiedContext;
+    private final Attributes modifiedAttributes;
+
+    static SpanData create(SpanData original) {
+        SpanContext originalSpanContext = original.getSpanContext();
+        Attributes attributes = original.getAttributes();
+        AttributeKey<String> traceIdKey = AttributeKey.stringKey("tmp.traceId");
+        AttributeKey<String> spanIdKey = AttributeKey.stringKey("tmp.spanId");
```

Comment:
Nit: WDYT about using a bit more descriptive names for the trace/span ID?
```suggestion
        AttributeKey<String> traceIdKey = AttributeKey.stringKey("_reactnative_traceId");
        AttributeKey<String> spanIdKey = AttributeKey.stringKey("_reactnative_spanId");
```
Also the underscore at the beginning makes it look a bit more "private"/"internal"